modules/test_infrastructure/main.tf

```diff
@@ -0,0 +1,214 @@
+
+  network_security_groups = each.value.network_security_groups
+  route_tables            = each.value.route_tables
+
+  tags = var.tags
+}
+
+# https://registry.terraform.io/modules/PaloAltoNetworks/swfw-modules/azurerm/latest/submodules/vnet_peering
+module "vnet_peering" {
+  source   = "../vnet_peering"
+  for_each = { for k, v in var.vnets : k => v if v.hub_vnet_name != null }
+
+
+  local_peer_config = {
+    name                = "peer-${each.value.name}-to-${each.value.hub_vnet_name}"
+    resource_group_name = local.resource_group.name
+    vnet_name           = each.value.name
+  }
+  remote_peer_config = {
+    name                = "peer-${each.value.hub_vnet_name}-to-${each.value.name}"
+    resource_group_name = try(each.value.hub_resource_group_name, local.resource_group.name)
+    vnet_name           = each.value.hub_vnet_name
+  }
+
+  depends_on = [module.vnet]
+}
+
+# https://registry.terraform.io/modules/PaloAltoNetworks/swfw-modules/azurerm/latest/submodules/loadbalancer
+module "load_balancer" {
+  source = "../../modules/loadbalancer"
+
+  for_each = var.load_balancers
+
+  name                = each.value.name
+  region              = var.region
+  resource_group_name = local.resource_group.name
+  zones               = each.value.zones
+  backend_name        = each.value.backend_name
+
+  health_probes = each.value.health_probes
+
+  nsg_auto_rules_settings = try(
+    {
+      nsg_name = try(
+        var.vnets[each.value.nsg_auto_rules_settings.nsg_vnet_key].network_security_groups[
+        each.value.nsg_auto_rules_settings.nsg_key].name,
+        each.value.nsg_auto_rules_settings.nsg_name
+      )
+      nsg_resource_group_name = try(
+        var.vnets[each.value.nsg_auto_rules_settings.nsg_vnet_key].resource_group_name,
+        each.value.nsg_auto_rules_settings.nsg_resource_group_name,
+        null
+      )
+      source_ips    = each.value.nsg_auto_rules_settings.source_ips
+      base_priority = each.value.nsg_auto_rules_settings.base_priority
+    },
+    null
+  )
+
+  frontend_ips = {
+    for k, v in each.value.frontend_ips : k => merge(
+      v,
+      {
+        public_ip_name = v.create_public_ip ? v.public_ip_name : null
+        subnet_id      = try(module.vnet[v.vnet_key].subnet_ids[v.subnet_key], null)
+        gwlb_fip_id    = try(v.gwlb_fip_id, null)
+      }
+    )
+  }
+
+  tags       = var.tags
+  depends_on = [module.vnet]
+}
+
+# https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/network_interface
+resource "azurerm_network_interface" "vm" {
+  for_each = var.spoke_vms
+
+  name                = each.value.interface_name
+  location            = var.region
+  resource_group_name = local.resource_group.name
+
+  ip_configuration {
+    name                          = "internal"
+    subnet_id                     = module.vnet[each.value.vnet_key].subnet_ids[each.value.subnet_key]
+    private_ip_address_allocation = "Dynamic"
```

Comment:
Do we want to support static IPs for test VMs e.g. if we want to be sure, that VM, for which we have NAT rule in VM-Series, has constant IP?

Reply:
Done.